jazz_services/index.js

```diff
@@ -73,45 +95,49 @@ module.exports.handler = (event, context, cb) => {
-                    crud.get(service_id, onComplete);
+                getServiceByServiceId: function(onComplete) {
+                    validateUtils.validateServiceWithServiceId(service_id, function onValidate(error, data) {
+                        onComplete(error, data);
+                    });
                 }
             }, function(error, data) {
                 if (error) {
                     logger.error('Error occured. ' + JSON.stringify(error, null, 2));
-                    return cb(JSON.stringify(errorHandler.throwInternalServerError('Unexpected Error occured.')));
-                }
-                var service_obj = data.getServiceByID;
-
-                // throw error if no service exists with given service_id
-                if (Object.keys(service_obj).length === 0 && service_obj.constructor === Object) {
-                    logger.error('Cannot find service with id: ' + service_id);
-                    return cb(JSON.stringify(errorHandler.throwNotFoundError('Cannot find service with id: ' + service_id)));
+                    
                 }
-                logger.verbose('Get Success. ' + JSON.stringify(service_obj, null, 2));
-                return cb(null, responseObj(data.getServiceByID, event.path));
+                if(data.getServiceByServiceId){
+                    var service_obj = data.getServiceByServiceId;
+                    logger.verbose('Get Success. ' + JSON.stringify(service_obj, null, 2));
+                    handleResponse(error, data.getServiceByServiceId, event.path)
+                } else{
+                    handleResponse(error, data, event.path);
+                }                
             });
         }
 
 
         // 2: GET all services (/services)
         // 3: GET Filtered services (/services?field1=value&field2=value2&...)
         if (event.method === 'GET' && !service_id) {
-            logger.info('GET services');
+            // logger.info('GET services');
             async.series({
                 // fetch services list from dynamodb, filter if required
                 fetchServices: function(onComplete) {
                     var query = event.query;
                     crud.getList(query, getAllRecords, onComplete);
                 }
             }, function(error, result) {
+                var data = result.fetchServices;
                 // Handle error
                 if (error) {
                     logger.error('Error occured. ' + JSON.stringify(error, null, 2));
-                    return cb(JSON.stringify(errorHandler.throwInternalServerError('unexpected error occured')));
+                    handleResponse(error, data.getServiceByServiceId, event.query)
                 }
-
-                return cb(null, responseObj(result.fetchServices, event.query));
+                if (data && data.services.length === 0 && event.query) {
+                    logger.error("Cannot find service with query: " + JSON.stringify(event.query));
```

Comment:
Why is this error? This is at best warn but mostly info

Reply:
changes are done as mentioned